core/services/blockhashstore/delegate.go

```diff
@@ -70,9 +71,9 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 	if len(keys) == 0 {
 		return nil, fmt.Errorf("missing sending keys for chain ID: %v", chain.ID())
 	}
-	fromAddress := keys[0].Address
-	if jb.BlockhashStoreSpec.FromAddress != nil {
-		fromAddress = jb.BlockhashStoreSpec.FromAddress.Address()
+	fromAddresses := []ethkey.EIP55Address{keys[0].EIP55Address}
```

Comment:
just as a note, this _might_ cause unexpected behavior if we misconfigure the job. The spec may still run but with a different key than we expect, which we will not be monitoring/topping up -> eventually it may run out of ETH and we wouldn't catch it missing blocks.

Reply:
Would maybe rather error out here than allow the job to start

Reply:
yeah good point. Perhaps that change can be made as a follow up because we have some job specs without any `fromAddress` today. Change will need to be made once all prod job specs have the sending keys explicitly set.

Reply:
> just as a note, this might cause unexpected behavior if we misconfigure the job.

>  Perhaps that change can be made as a follow up because we have some job specs without any fromAddress today.

I think we should sweep all existing jobs that we have in prod and specify a `fromAddress` explicitly before the migration to be 100%.